third-party/crowdsec/src/test/java/org/apache/james/CrowdsecHttpClientTest.java

```diff
@@ -0,0 +1,65 @@
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james;
+
+import static org.apache.james.model.CrowdsecClientConfiguration.DEFAULT_API_KEY;
+import static org.apache.james.model.CrowdsecDecision.BAN;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.List;
+
+import org.apache.james.model.CrowdsecClientConfiguration;
+import org.apache.james.model.CrowdsecDecision;
+import org.apache.james.model.CrowdsecHttpClient;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+class CrowdsecHttpClientTest {
+    @RegisterExtension
+    static CrowdsecExtension crowdsecExtension = new CrowdsecExtension();
+
+    @Test
+    void getDecisionsWhenBanningAnIP() throws IOException, InterruptedException {
+        crowdsecExtension.getCrowdsecContainer().execInContainer("cscli", "bouncer", "add", "bouncer", "-k", DEFAULT_API_KEY);
+        crowdsecExtension.getCrowdsecContainer().execInContainer("cscli", "decision", "delete", "--all");
+        crowdsecExtension.getCrowdsecContainer().execInContainer("cscli" , "decision", "add", "-i", "192.168.0.4");
+        CrowdsecClientConfiguration config = new CrowdsecClientConfiguration(new URL("http://localhost:8082/v1"), DEFAULT_API_KEY);
+        CrowdsecHttpClient httpClient = new CrowdsecHttpClient(config);
+        List<CrowdsecDecision> decisions = httpClient.getCrowdsecDecisions().block();
+        assert decisions != null;
+        assertThat(decisions).hasSize(1);
+        assertThat(decisions.get(0).getValue().equals("192.168.0.4") && decisions.get(0).getType().equals(BAN));
+    }
+
+    @Test
+    void getDecisionsWhenBanningAnIPRange() throws IOException, InterruptedException {
+        crowdsecExtension.getCrowdsecContainer().execInContainer("cscli", "bouncer", "add", "bouncer", "-k", DEFAULT_API_KEY);
+        crowdsecExtension.getCrowdsecContainer().execInContainer("cscli", "decision", "delete", "--all");
+        crowdsecExtension.getCrowdsecContainer().execInContainer("cscli" , "decision", "add", "-r", "192.168.0.0/16");
+        CrowdsecClientConfiguration config = new CrowdsecClientConfiguration(new URL("http://localhost:8082/v1"), DEFAULT_API_KEY);
+        CrowdsecHttpClient httpClient = new CrowdsecHttpClient(config);
+        List<CrowdsecDecision> decisions = httpClient.getCrowdsecDecisions().block();
+        assert decisions != null;
+        assertThat(decisions).hasSize(1);
+        assertThat(decisions.get(0).getValue().equals("192.168.0.0/16") && decisions.get(0).getType().equals(BAN));
+    }
```

Comment:
Please test more carefully.

Any other tests for failure cases? e.g.:
- wrong CrowdSec url
- wrong CrowdSec apikey
- response when no ban decision
- response when multiple ban decisions
...